sympy/core/symbol.py

```diff
@@ -107,7 +106,19 @@ def __new_stage2__(cls, name, **assumptions):
 
         obj = Expr.__new__(cls)
         obj.name = name
+
+        # FIXME: bit horrid: to avoid having "commutative=True" show up in
+        # Symbol's srepr, keep a copy...
+        tmpasmcopy = assumptions.copy()
+
+        # be strict about commutativity
+        is_commutative = fuzzy_bool(assumptions.get('commutative', True))
+        assumptions['commutative'] = is_commutative
         obj._assumptions = StdFactKB(assumptions)
+
+        # FIXME: ... then use the copy overwrite the generator
+        obj._assumptions._generator = tmpasmcopy
+
```

Comment:
> Let's go with this solution, but just note in the code that it is a hack

Ok, do you feel this current note is ok?  Do you want it to refer to this issue #?  Or file some other issue now and point it there before we merge this?